src/Streams.Core/Streams.fs

```diff
@@ -4,20 +4,18 @@ open System.Collections.Generic
 open System.Threading
 
 /// Provides on-demand iteration 
-type Iterator = {
+type (* internal *) Iterator = 
     /// Function for on-demand processing
-    TryAdvance : unit -> bool 
+    abstract TryAdvance : unit -> bool 
     /// Cleanup function
-    Dispose : unit -> unit 
-}
+    abstract Dispose : unit -> unit 
 
 /// Provides functions for iteration
-type Iterable = {
+type (* internal *) Iterable = 
```

Comment:
According to this https://github.com/nessos/Streams/blob/master/src/Streams.Core/Streams.fs#L7-L18
Iterable is like IEnumerable plus bulk (which by construction it has scope and can do its own cleanup) and Iterable which is like IEnumerator minus the Pull based Current property.